cmd/kured/main.go

```diff
@@ -333,19 +342,28 @@ func drain(client *kubernetes.Clientset, node *v1.Node) {
 
 	drainer := &kubectldrain.Helper{
 		Client:              client,
-		GracePeriodSeconds:  -1,
+		Ctx:                 context.Background(),
+		GracePeriodSeconds:  drainGracePeriod,
 		Force:               true,
 		DeleteEmptyDirData:  true,
 		IgnoreAllDaemonSets: true,
 		ErrOut:              os.Stderr,
 		Out:                 os.Stdout,
+		Timeout:             drainTimeout,
 	}
+
 	if err := kubectldrain.RunCordonOrUncordon(drainer, node, true); err != nil {
-		log.Fatalf("Error cordonning %s: %v", nodename, err)
+		if !forceReboot {
+			log.Fatalf("Error cordonning %s: %v", nodename, err)
+		}
+		log.Errorf("Error cordonning %s: %v, continuing with reboot anyway", nodename, err)
```

Comment:
Is there a side effect possible:
- when forceReboot
- when the cordon has failed
- and then the drain is triggered ?

Reply:
it's unlikely, but just to be safe, I added an explicit return after the `Errorf` log statement.

Reply:
Sorry this is not what I meant.

I meant that `drain` itself could be refactored to bubble up errors if necessary. Instead of having the current logic:
`if not force, fatal, else log error` in the current code, we could make sure drain returns an error, and the `rebotoAsRequired` could then behave differently, based on the error.

FYI it's a nit. It's food for thought. You don't need to change your current code, as it does what you tell it does.
What I mentioned can be done in a later patch. I just think it's probably easier to read that way. WDYT?

Reply:
> Sorry this is not what I meant. I meant that I don't really know the details behind this kubernetes behaviour.

Oh I see. I assumed you meant draining when the cordon failed could lead to unexpected states.

I agree the error handling is all over the place here. The `forceReboot` condition makes the error handling even harder to follow. I'm not against rewriting `drain` to return an error if you think its in the scope of this PR.

Reply:
Let's do that in another PR :) 

I would be happy to see a follow up PR from you!